client/ayon_applications/ui/process_monitor.py

```diff
@@ -0,0 +1,991 @@
+                file_deleted = 1
+        # Delete from database
+        deleted = self._manager.delete_process_info(self._process_hash)
+        process_deleted = 1 if deleted else 0
+
+        self.signals.finished.emit(process_deleted, file_deleted)
+
+
+class ProcessTreeModel(QtGui.QStandardItemModel):
+    """Model for displaying process information.
+
+    Each row represents a ProcessInfo. ProcessInfo objects are stored in
+    Qt.UserRole on the first item of the row for easy retrieval.
+    """
+
+    def __init__(
+            self,
+            parent: Optional[QtCore.QObject] = None,
+            manager: Optional[ApplicationManager] = None) -> None:
+        """Initialize the model.
+
+        Args:
+            parent (Optional[QtCore.QObject]): Parent QObject.
+            manager (Optional[ApplicationManager]): Application manager
+
+        """
+        super().__init__(parent)
+        self._processes: list[ProcessInfo] = []
+        self._manager = manager
+        # Columns
+        self.headers = [
+            "Name", "Executable", "PID", "Status", "Created", "Start Time",
+            "Site ID", "Output File", "Hash"
+        ]
+        self.columns = enum.IntEnum(  # type: ignore[misc]
+            "columns",
+            {
+                name.replace(" ", "_").upper(): i
+                for i, name in enumerate(self.headers)
+            },
+        )
+        self.setColumnCount(len(self.headers))
+        self.setHorizontalHeaderLabels(self.headers)
+
+    @staticmethod
+    def _status_icon(process: ProcessInfo, size: int = 12) -> QtGui.QIcon:
+        """Return a small colored circle icon representing process status.
+
+        Args:
+            process (ProcessInfo): ProcessInfo object.
+            size (int): Size of the icon in pixels.
+
+        Returns:
+            QtGui.QIcon: Colored circle icon.
+
+        """
+        pix = QtGui.QPixmap(size, size)
+        pix.fill(QtCore.Qt.GlobalColor.transparent)
+        painter = QtGui.QPainter(pix)
+        painter.setRenderHint(QtGui.QPainter.RenderHint.Antialiasing)
+        if process.pid and process.active:
+            color = QtGui.QColor(0, 180, 0)  # green = running
+        elif process.pid:
+            color = QtGui.QColor(200, 0, 0)  # red = stopped
+        else:
+            color = QtGui.QColor(140, 140, 140)  # gray = unknown
+        painter.setBrush(QtGui.QBrush(color))
+        painter.setPen(QtCore.Qt.PenStyle.NoPen)
+        painter.drawEllipse(1, 1, size - 2, size - 2)
+        painter.end()
+        return QtGui.QIcon(pix)
+
+    def update_processes(self, processes: list[ProcessInfo]) -> None:
+        """Replace current content with provided processes.
+
+        Args:
+            processes (list[ProcessInfo]): List of ProcessInfo objects.
+
+        """
+        self._processes = list(processes)
+        self.clear()
+        self.setColumnCount(len(self.headers))
+        self.setHorizontalHeaderLabels(self.headers)
```

Comment:
`clear` resets the model to "initial state", not just rows but also headers and every other thing.
```suggestion
        root_item = self.invisibleRootItem()
        root_item.removeRows(0, root_item.rowCount())
```

Reply:
Additional note:
It would be greate to NOT remove the items, but only update data of existing ones. By removing them you also loose any UI track of the process items (like selection).